File: backend/src/main/java/com/ssafy/zip/entity/Calendar.java
package com.ssafy.zip.entity;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import lombok.*;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Table(name = "calendar")
public class Calendar {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long id;

    LocalDateTime startDate;

    LocalDateTime endDate;

    String schedule;

    @JsonManagedReference
    @OneToMany(mappedBy = "calendar")
    List<CalendarUser> calendarUsers;

    @ManyToOne
    @JoinColumn(name = "family_id")
    Family family;

    public void editCalendar(LocalDateTime startDate, LocalDateTime endDate, String content){
        this.startDate = startDate;
        this.endDate = endDate;
        this.schedule = content;
    }
}

File: android/app/src/main/java/com/ssafy/zip/android/data/Calendar.kt
package com.ssafy.zip.android.data

import android.os.Parcelable
import kotlinx.parcelize.Parcelize
import java.time.LocalDate
import java.util.*

@Parcelize
data class Calendar(
    val content: String,
    val endDate: Date,
    val id: Long,
    val startDate: Date,
    val users: ArrayList<FamilyMember>

) : Parcelable
